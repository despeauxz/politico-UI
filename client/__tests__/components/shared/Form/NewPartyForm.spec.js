import React from 'react';
import NewPartyForm from '@components/shared/Form/Party/NewParty';

const props = {
    loading: false,
    createParty: jest.fn()
};

describe('New Party Form', () => {
    it('should render the inew party modal', () => {
        const shallowWrapper = shallow(<NewPartyForm {...props} />);

        expect(toJson(shallowWrapper)).toMatchSnapshot();
        expect(shallowWrapper.find('NewPartyForm')).toBeTruthy();
    });
});