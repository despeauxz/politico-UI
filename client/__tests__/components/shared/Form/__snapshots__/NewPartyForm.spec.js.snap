// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Party Form should render the inew party modal 1`] = `
<Fragment>
  <form
    className="mt-6"
    onSubmit={[Function]}
  >
    <RenderInput
      className="form-control border border-indigo-700 text-gray-600"
      error=""
      errors={Object {}}
      handleChange={[Function]}
      id="name"
      label="Name"
      name="name"
      onFocus={[Function]}
      placeholder="Name"
      type="text"
      value=""
    />
    <RenderInput
      className="form-control border border-indigo-700 text-gray-600"
      error=""
      errors={Object {}}
      handleChange={[Function]}
      id="fullname"
      label="Full Name"
      name="fullname"
      onFocus={[Function]}
      placeholder="Full Name"
      type="text"
      value=""
    />
    <RenderInput
      className="form-control border border-indigo-700 text-gray-600"
      error=""
      errors={Object {}}
      handleChange={[Function]}
      id="hqAddress"
      label="HQ Address"
      name="hqAddress"
      onFocus={[Function]}
      placeholder="HQ Address"
      type="text"
      value=""
    />
    <div
      className="flex items-center justify-center"
    >
      <label
        className="w-64 flex flex-col items-center px-2 py-2 bg-white text-blue rounded-lg shadow-lg tracking-wide uppercase border border-blue cursor-pointer hover:bg-blue-600 hover:text-white"
      >
        <svg
          className="w-4 h-4"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.88 9.1A4 4 0 0 1 16 17H5a5 5 0 0 1-1-9.9V7a3 3 0 0 1 4.52-2.59A4.98 4.98 0 0 1 17 8c0 .38-.04.74-.12 1.1zM11 11h3l-4-4-4 4h3v3h2v-3z"
          />
        </svg>
        <span
          className="mt-2 text-base leading-normal"
        >
          Select a file
        </span>
        <input
          className="hidden upload-button"
          onChange={[Function]}
          type="file"
        />
      </label>
    </div>
    <div
      className="flex items-center justify-center"
    />
    <button
      className="py-3 px-6 my-8 rounded text-white bg-teal-600 hover:bg-teal-700 shadow-lg"
      disabled={false}
      type="submit"
    >
      <span
        className="inline-block mr-2"
      >
        <Plus />
      </span>
       Create
    </button>
  </form>
</Fragment>
`;
